### ISSUE TITLE
Few small quirks using Windows 10

### ISSUE THREAD

**Issue by Gavin-Holt** [rxi/lite#38]
Hi

I am really enjoying exploring Lite (Windows 10 - on a small screen laptop). The speed and beauty are very attractive, but there are a few minor points that detract from my pleasure.

1. Rendering long lines

I respect the decision against proper word wrapping, but the current situation is not "nice". As I type near the right window margin the whole screen scrolls, with no horizontal scroll bar - this is very disorientating.

When the screen is in this scrolled state I cannot accurately select a position with the mouse - the text jumps left/right - a single click selects several characters.

2. core.command_view:enter leaves pixels on the last row of the screen

On first opening there is a very clean window. After each use of command entry / find / eval / exec - I can see pixels on the last row of the screen - matching those from the top row of the single line command window which has been closed (?off by one). These stray pixels are unaffected by window scrolling or resizing.

3. Log view won't scroll to the right

If I mess up and Lua issues a long error message I can't see the text beyond the right screen edge. The End and Right keys don't function, and there is no horizontal scroll bar.

4. Eval text entry not syntax highlighted

I would be really nice to have Lua syntax highlighting when typing in the eval plugin.

5. Output Window 

When using the eval plugin (windows 10) any print statements produce no output. The return value inserts as expected. Is there any way to capture stdout and append to a buffer? I would prefer a solution with output in a fully editable buffer - not a limited buffer like "Log". 

6. Loading binary libraries

I note that package.cpath is set (e.g. P:\MyPrograms\EDITORS\LiteMOD\?.dll;P:\MyPrograms\EDITORS\LiteMOD\loadall.dll;.\?.dll) but I can't get any of my Lua52 binary libraries to load. I understand some may be dependent upon lua52.dll but I can't get anything to load. (Any examples that work would be interesting - so I can see what was used to compile them)

Many thanks for the latest release, especially for eval which allows me to prod and poke.

Kind Regards Gavin Holt 

I am using the latest release, and the _SCALE variable in my init.lua to make things bigger.

**Comment by rxi** [comment]
Thanks for the kind words, regarding your points:
1. What do you suggest might be a better solution than the current one? The issue you mentioned with the selecting-several-characters on a single fix has been fixed on the master branch (thus the next release will have this fix)
2. Would you be able to make a gif (or screenshot) of this and let me know your scale / settings so I can try and reproduce it. I sorted an issue on the master branch regarding the CommandView's floating window height, but this sounds like a separate issue
3. I'd noticed this becomes a bit of an issue with long absolute filenames, not yet settled on an elegant solution for it, though
4. This *could* be done but adding this functionality just for this use case would be a bit overkill
5. You could edit the plugin to patch the global `print` function (or pass it an environment with `print` set as a different function) that writes to a buffer instead of actually printing. This isn't functionality I want to support in the standard `eval` plugin, though. *(this and the previous point make me think you may be using the `eval` plugin for something more than [it was intended](https://user-images.githubusercontent.com/3920290/80301638-0dd68800-879d-11ea-8d38-85030fbb6a54.gif))*.
6. Do you get the error: `dynamic libraries not enabled; check your Lua installation`? If so this should be an easy fix

**Comment by Gavin-Holt** [comment]
Hi

Many thanks for such a prompt reply. 

1. I think a horizontal scroll bar would help with orientation. Great to hear the selection problem will be fixed - is this a lua fix I can apply or do I need to wait for the next compiled exe?

2. I am using summer colours and scaling factor applied in my init file. Its just a few pixels at the very bottom
![OnLoad](https://user-images.githubusercontent.com/5831149/80310497-ee5a5200-87d2-11ea-80b5-549e9b5d3fdd.png)
![AfterDoCommand](https://user-images.githubusercontent.com/5831149/80310499-ef8b7f00-87d2-11ea-91ac-aeee158d1bdb.png)

3. A horizontal scroll bar would also answer this problem :)

4. I fully understand changing core.command_view:enter may be overkill. I did notice that CommandView is implemented using SingleLineDoc = Doc:extend(). Could the input to eval use a single line doc where highlighting might be much easier to apply?

5. I admit it I would like to use eval for more than simple single function commands. Could we use eval to access the text processing capabilities of Lua (see https://foicica.com/lists/code/202002/5380.html - Sorry about the lack of line spacing) 

TLDR; I would like to be able to input para():gsub("Python","Lua"):ins() to make a paragraph selection, a substitution and an undo-able insert. With a few more string functions you quickly have a very comprehensive editing language. You won't be surprised if my next task is to remember the eval strings and create a hotkey to "repeat last". 

6. Every time I try to require a DLL I get the following error:
![RequireDLLError](https://user-images.githubusercontent.com/5831149/80311150-db954c80-87d5-11ea-8635-d106692a6a07.png)

..and the rest of the init script does not run.

I would take this opportunity to thank you for this software, I love exploring new editors and I am sorry my skills don't allow me to assist in the C coding (I'm a knee surgeon by profession)

Kind Regards Gavin Holt

**Comment by rxi** [comment]
1. I don't plan to add horizontal scrollbars, it took a while before I even committed to having the sparsely used vertical ones! `v1.02` on the releases page fixes the selection problem
2. It took me a moment to notice you removed (or hid) the StatusView! This isn't really "supported" behaviour and the reason for the bug would depend on how you did this. In general all space needs to be accounted for, there isn't a default "background" thus, in a case like this, if an area becomes unaccounted for -- that is, isn't being drawn over by a View or a divider -- it will just remain as it was the previous frame. As hiding the StatusView isn't technically supported I won't be looking into this part further
3. A horizontal scrollbar would be a *bad solution* to this problem. Given there's no rush to improve this there's no need to solve it in a less-than-ideal way
4. The CommandView is just a `DocView`, thus you could just manually load a syntax into it (see `reset_syntax()` in `docview.lua`) -- that being said, again, this feels like misuse of it
5. Skimmed the link -- again this feels like misuse of this specific plugin. That isn't to say the functionality shouldn't be supported in a different plugin, it just doesn't seem to relate to this one. If you want to add additional text processing stuff see some of the other plugins that do this, eg. all the Doc commands in `commands/doc.lua` or something like the `reflow` plugin
6. I think that specific error was caused by a bug I had in `core.try` (if you update your lua files from the master branch this should be fixed). My assumption is that it will then show you the correct error and that this should be a trivial fix which I will look into when I get a chance

**Comment by Gavin-Holt** [comment]
Many thanks for your help. I will try the new Lua files.

**Comment by SwissalpS** [comment]
>     6\. Do you get the error: `dynamic libraries not enabled; check your Lua installation`? If so this should be an easy fix

I do when I attempt to load a lua module from within a plugin. What is the easy fix? Some compiler directive?

I'm using Fedora and would like to load installed https://github.com/Tangent128/luasdl2 (as example)
